autoload/ale/lsp_linter.vim

```diff
@@ -271,6 +271,20 @@ function! ale#lsp_linter#OnInit(linter, details, Callback) abort
         call ale#lsp#NotifyForChanges(l:conn_id, l:buffer)
     endif
 
+    " Tell the relevant buffer that the LSP has started.
+    if l:buffer > 0 && l:buffer == bufnr('')
+        silent doautocmd <nomodeline> User ALELSPStarted
+    else
+        execute 'augroup ALELSPStartedGroup' . l:buffer
+            autocmd!
+
+            execute printf(
+            \   'autocmd BufEnter <buffer=%d> User ALELSPStarted',
```

Comment:
This looks wrong. It doesn't use `doautocommand`.

It also doesn't get removed once fired. 

See https://github.com/prabirshrestha/vim-lsp/blob/fb0a72306e1fc42c1ce8909aa1e420ddfccf6fbf/autoload/lsp.vim#L345 for one way to approach it. 

The test likely doesn't see it because it doesn't switch buffers after asking the LSP to start but before this code-path is hit.

Reply:
Yeah, I totally lost the plot on this branch when I started reading about `++once` and didn't come back to finish it up.